### PR TITLE
SBERDOMA-873 configure apollo logging

### PR DESCRIPTION
**Pull request by MrFoxPro**

### CODE REVIEW

File: apps/condo/domains/common/utils/apolloErrorFormatter.js [hunk 1]
```diff
@@ -123,27 +123,27 @@ const toGraphQLFormat = (safeFormattedError) => {
...
-        // NOTE1(pahaz): Keystone use apollo-errors for all their errors. There are:
-        //   AccessDeniedError, ValidationFailureError, LimitsExceededError and ParameterError
-        // NOTE2(pahaz): Apollo use apollo-server-errors for all their errors> There are:
-        //   SyntaxError, ValidationError, UserInputError, AuthenticationError, ForbiddenError, PersistedQueryNotFoundError, PersistedQueryNotSupportedError, ...
-        if (isKeystoneErrorInstance(originalError) || originalError instanceof ApolloError) {
+            // NOTE1(pahaz): Keystone use apollo-errors for all their errors. There are:
+            //   AccessDeniedError, ValidationFailureError, LimitsExceededError and ParameterError
+            // NOTE2(pahaz): Apollo use apollo-server-errors for all their errors> There are:
+            //   SyntaxError, ValidationError, UserInputError, AuthenticationError, ForbiddenError, PersistedQueryNotFoundError, PersistedQueryNotSupportedError, ...
+            if (isKeystoneErrorInstance(originalError) || originalError instanceof ApolloError) {
```
Comment (AntonAL) [inline apps/condo/domains/common/utils/apolloErrorFormatter.js:135]: Sources are not a conversation place ;)
Please, write a generic comment

File: packages/@core.config/index.js [hunk 2]
```diff
@@ -37,7 +37,10 @@ if (DEBUG) console.log(`@core.config: inited! namespace=${namespace}, cwd=${cwd}
 if (DEBUG) console.dir(process.env)
 
 function getEnv (namespace, name, defaultValue) {
-    return preprocessEnv(process.env[`${namespace}_${name}`] || process.env[`${name}`] || defaultValue)
+    let variable = preprocessEnv(process.env[`${namespace}_${name}`] || process.env[`${name}`] || defaultValue)
```
Comment (AntonAL) [inline packages/@core.config/index.js:40]: Please, give to `variable` a meaningful name ;)

**Review by AntonAL — CHANGES_REQUESTED** [review]
Please, create a `apps/condo/docs/debugging.md` file and describe the usage of `DEBUG_APOLLO` variable and how it will help to debug the application.
Do I understand correctly, that with this variable we get extra `uid` field in error?

File: apps/condo/index.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ const bodyParser = require('body-parser')
 const { formatError } = require('@condo/domains/common/utils/apolloErrorFormatter')
 
 const IS_ENABLE_DD_TRACE = conf.NODE_ENV === 'production'
-const IS_ENABLE_APOLLO_DEBUG = conf.NODE_ENV === 'development' || conf.NODE_ENV === 'test'
+const IS_ENABLE_APOLLO_DEBUG = conf.DEBUG_APOLLO && conf.NODE_ENV !== 'production'
```
Comment (AntonAL) [inline apps/condo/index.js:21]: For what use-case Apollo debugging mode is enabled only for non-production environment?
How that extra `uid` field will help me to debug an application in `development` and `test` node environments?

Comment (MrFoxPro) [reply apps/condo/index.js:21]: @AntonAL 
Whad do you mean about uid?

**Review by Dimitreee — APPROVED** [review]

File: .env.example [hunk 4]
```diff
@@ -5,6 +5,7 @@ NODE_ENV=development
 DISABLE_LOGGING=true
 COOKIE_SECRET=random
 SERVER_URL=http://localhost:3000
+DEBUG_APOLLO=false
```
Comment (pahaz) [inline .env.example:8]: It's a bad idea to hardcode the installation name inside ENV.

File: apps/condo/domains/common/utils/apolloErrorFormatter.js [hunk 5]
```diff
@@ -123,27 +123,26 @@ const toGraphQLFormat = (safeFormattedError) => {
...
-        //   AccessDeniedError, ValidationFailureError, LimitsExceededError and ParameterError
-        // NOTE2(pahaz): Apollo use apollo-server-errors for all their errors> There are:
-        //   SyntaxError, ValidationError, UserInputError, AuthenticationError, ForbiddenError, PersistedQueryNotFoundError, PersistedQueryNotSupportedError, ...
-        if (isKeystoneErrorInstance(originalError) || originalError instanceof ApolloError) {
-            // originalError: { message name data internalData time_thrown path locations }
-            graphqlLogger.info({ error: safeFormatError(error) })
-        } else {
-            graphqlLogger.error({ error: safeFormatError(error) })
+    error.uid = cuid()
+    if (conf.DEBUG_APOLLO){
```
Comment (pahaz) [inline apps/condo/domains/common/utils/apolloErrorFormatter.js:127]: it's a bad idea to use a stand-specific config. It's a bad practice

File: .env.example (hunk 4 above)
Comment (pahaz) [inline .env.example:8]: It's a bad practice to use environment-dependent code/logic.

**Review by pahaz — CHANGES_REQUESTED** [review]

**Comment by MrFoxPro** [comment]
The is another issue. This problem was partially solved by #400

**Pull request closed without merging**